### PR TITLE
Tickets/dm 28626

### PR DESCRIPTION
**Pull request by iagaponenko**

### CODE REVIEW

File: src/replica/Configuration.cc [hunk 1]
```diff
@@ -735,20 +742,52 @@ DatabaseInfo& Configuration::_databaseInfo(util::Lock const& lock, string const&
...
+        throw logic_error(_context(__func__) + " database '" + databaseInfo.name + "' is already published.");
+    } else if (!publish && !databaseInfo.isPublished) {
+        throw logic_error(_context(__func__) + " database '" + databaseInfo.name + "' is not published.");
+    }
+    if (publish) {
+        uint64_t const publishTime = PerformanceUtils::now();
+        // Firstly, publish all tables that were not published.
+        for (auto const& table : databaseInfo.tables()) {
+            if (_connectionPtr != nullptr) {
+                _connectionPtr->executeInOwnTransaction([&](decltype(_connectionPtr) conn) {
```
Comment (n8pease) [inline src/replica/Configuration.cc:749]: default capture modes of lambda expressions make me nervous.
https://learning.oreilly.com/library/view/effective-modern-c/9781491908419/ch06.html#:-:text=Item%2031%3A%E2%80%82Avoid%20default,may%20not%20be).
(you can log in with your Stanford credentials)
YMMV.

File: src/replica/IngestRequest.cc [hunk 2]
```diff
@@ -421,6 +419,7 @@ void IngestRequest::_readLocalFile() {
         raiseRetryAllowedError(context, "failed to open the file '" + _resource->filePath() + "', error: '" +
                                                 strerror(errno) + "', errno: " + to_string(errno));
     }
+    unique_ptr<csv::Parser> parser(new csv::Parser(_dialect));
```
Comment (n8pease) [inline src/replica/IngestRequest.cc:422]: use make_unique?

Comment (n8pease) [reply src/replica/IngestRequest.cc:422]: and on line 477

**Comment by n8pease** [comment]
It would be a little bit better for the git history if you rebased the clang-format changes back into their respective commits.

**Review by fritzm — APPROVED** [review]
LGTM, thanks!

File: src/replica/IngestRequest.cc (hunk 2 above)
Comment (iagaponenko) [reply src/replica/IngestRequest.cc:422]: Good point.

**Pull request merged**